### PR TITLE
ci(overhaul): waves A+B in shadow, plus licensing Wave 1 (total tier map, enforcement battery, fail-closed skips)

### PR DESCRIPTION
**Pull request by mfbayraktar**
## What this is

Two waves in one branch. The CI-overhaul repair work, and Wave 1 of the
config-universe testing pillar. The stack that was on `0728-3` is folded in
here rather than opened as a second PR.

## Submodule PRs

These must land before this PR, and the pointers here must then be re-pointed
at their squash commits. Dropping these links is what failed
`Quality / Submodule Branches` on the previous run.

- https://github.com/rediacc/renet/pull/95
- https://github.com/rediacc/elite/pull/14
- https://github.com/rediacc/account/pull/71

## CI overhaul

- **Nightly resurrection.** The watchdog no longer force-cancels a `schedule`
  run, so a failed nightly reads as `failure` instead of `cancelled`. That
  masking is what hid twelve consecutive red nights. A red nightly now raises
  an issue.
- **Scope engine**, landed in SHADOW and gating nothing: `--classify` plus
  `--resolve-baseline`, which finds the last green FULL run so a small push
  after a green round can reduce rather than re-diffing the whole branch.
  Fails open to full CI on every defect, each with a stated reason.
- **Skip-plan reconciliation**, also shadow. It runs the real reconciler
  against real per-job outcomes and reports what it WOULD have done, because
  wiring it live today would red every push-to-main.
- **Job-aggregation, swallowed-failure and scope gates**, each with a planted
  defect proving it can fire.

Two planned items were **dropped after measurement**: the macOS demotion
(queues in 0.4 min, finishes with 42.5 min of slack) and the D5 tag split (the
last Build job ends at +18 min on 55 to 61 minute runs). Both were justified by
figures from a contaminated attempt-2 run.

## Licensing (config-universe Wave 1)

- **T1**: an enforcement battery against a per-run ephemeral key. 14 scenarios,
  no VM, no BTRFS, no Docker. Two controls, both required to fail: a
  `-tags nolicense` build and a wrong-key build.
- **T2**: the tier map is now TOTAL. It was a switch whose `default: TierNone`
  made "nobody classified this" indistinguishable from "decided: no licence
  needed" for 154 of 162 registered functions. A second hidden decision point
  (a `repository_` prefix filter) is removed, so the map is the only place a
  decision lives.
- **T3**: the map is emitted into the generated TypeScript contract, and the
  CLI's hand-maintained duplicate is deleted. That duplicate had already
  drifted, claiming every `backup_*` function was licensed when renet licenses
  none.
- **T4**: the subscription system tests now FAIL when silently omitted. A job
  with no account server must declare it.
- **Dispatch tests** proving the map actually drives enforcement, including
  that `repository_down`/`repository_delete` stay ungated, which is the pricing
  page's promise that stop and delete keep working after a subscription ends.

## Defects found and fixed on the way

Not part of the original scope; found while building the above and fixed rather
than filed, per the repo's standing rule.

- `build-renet.sh` skipped the build whenever anything existed in `bin/`,
  regardless of how it was built. **Ten CI steps run it**, so a job that built
  `--nolicense` followed by one wanting an enforcing binary silently reused the
  nolicense one and its licence assertions passed for free. Now keyed on a build
  identity (flag + env opt-in + hash of the baked key).
- `build.sh` baked the **production** key from `private/account/.env` with no
  indication when `ACCOUNT_ED25519_PUBLIC_KEY` was unset. Behaviour unchanged,
  now warns with the key's fingerprint (never the key).
- `concurrency.test.ts` was named "should allow only maxActivations concurrent
  machines" while asserting `<= 2` with a cap of 1. Renamed to what it proves;
  deliberately not tightened, because better-sqlite3 is synchronous so a strict
  assertion would pass locally while the cap stays violable on D1.
- `run-tests.sh` failed `shellcheck -S warning` (SC1090) on the committed
  version, invisible because renet's CI runs no shellcheck.

- The shadow observers wrote only to `$GITHUB_STEP_SUMMARY`, which has no API,
  so their verdicts were unreadable by anything automated. Now tee'd to stdout.
- The reconciler reported `planned-job-missing` for a skipped reusable caller
  (GitHub materialises only `OPS Tests`, never its inner jobs). Now reported as
  `planned-run-but-skipped`, which is what actually happened.

- `validate-pr.cjs` read the PR from the frozen event payload, so a title fix
  could never be validated by a re-run: it replayed the old title. Proven live
  (run 30393629318 failed, title corrected, re-run failed identically). Now
  reads the PR through the API, with the payload as fallback.

Related: #533, #534, #537, #538, #539.

- The external-link gate reported "All external links are valid" against a tree
  with no docs directory: a hardcoded `DOCS_DIR`, zero files, zero iterations,
  a checkmark. Found by planting a dead URL to prove the planalto.gov.br fix and
  watching nothing happen. It now refuses an empty scan set and is registered in
  `test-gate-anti-vacuity.sh`, which proves it fires.

- Both Claude review findings fixed, each verified against the artifact first:
  the licence battery's `chmod -R` covered the whole store instead of the repo
  directory it wrote (harmless on CI, permanent on a dev machine with real
  licences), and the reconciler's caller-derivation blamed the `Tests + Infra`
  caller when the flat `package-tests` job went missing. Case 33 added, with a
  control proving a genuine reusable caller still diagnoses as a caller skip.

- The preview readiness gate waited on `/health` while the smoke test's first
  real call is `/.well-known/server-info`, which has a different dependency
  footprint. Round 9 printed "Preview worker is ready" and then took HTTP 500
  one line later. The gate now requires both, and requires server-info to carry
  a usable body. Proven with a control: replacing the server-info marker fails
  the gate while `/health` is up, so the new condition is load-bearing.

- Four items that had been sitting on a "found, not fixed" list are fixed:
  `parse_args` assigned with `eval`, so an argument value containing `;` or
  `$(...)` was executed rather than stored (proven by driving the real function,
  reverted-and-reproduced, restored by sha256); `detect-pointer-bump.sh`
  resolved head with `git rev-parse HEAD`, which on a `pull_request` is the
  synthetic two-parent merge ref, so the walk aborted every time and
  `pointer_bump_only` was false 13 runs for 13; and two comments that stated
  intent rather than measured behaviour (the cron claims 01:00 but has never
  dispatched before 03:53, and the retry allowlist bounds the trigger, not the
  blast radius).

- Three dependencies aged past the 24-hour freshness window and the gate demanded
  them (`eslint-plugin-playwright`, `@marsidev/react-turnstile`,
  `@modelcontextprotocol/sdk`). Taken via `check:deps --upgrade`, then the
  lockfile normalised with `npm@10` because the system npm is 11 and produces the
  cosmetic dev-marker flip. Verified rather than assumed: 20/20 lockfile lines
  changed with **zero** dev-marker lines touched.

- The documented licence-repro command was silently poisoning the build:
  `ACCOUNT_ED25519_PUBLIC_KEY="$(curl -s .../api/public/account-key)"` hits a 404,
  and `curl -s` exits 0, so an HTML error page became the key and got baked into
  `keys.ProductionPublicKey` via ldflags. Every prod-signed licence then fails as
  `invalid_signature` - the exact symptom that variable exists to prevent. Both
  call sites now use `-fsS` (proven: exit 0 vs exit 22). The URL is flagged, not
  guessed: the account API answers `410 Gone` on www.rediacc.com.

- **Security**: an earlier commit in this PR (680203a49) persisted credentials on
  the review's PR-head checkout. That writes the raw bearer token into
  `.git/config` in a workspace holding PR-authored content, where the model has
  `Read` plus `gh pr comment`/`gh api` egress under `pull-requests: write`.
  Fixed at the source: forcing `TRACK_PROGRESS: 'false'` means the action never
  fetches, so no credential is needed and `persist-credentials: false` is
  restored. A tripwire step now fails the job if one is ever re-introduced
  (proven able to fire). Also adds producer-closure image tags (`--closure
  web|rdc`), built and proven but deliberately NOT wired, since switching the
  live tags is release-affecting.

- **D5 wired**: `WEB_TAG` and `RDC_TAG` now hash each image's producer closure
  instead of the console commit, so they are finally disjoint. Bounded by a
  weekly ISO bucket, because a closure key can outlive its inputs where a commit
  key cannot, and it provably does not cover the floating base images or
  `private/account`'s unpinned `npm install`. Acceptance case re-run against the
  wired path: the `.dockerignore`d file from 8b7840ed4 moves `--self` and leaves
  both closure tags untouched.
- Also fixes `cancel-old-ci.sh`, which had been a no-op for this entire wave: it
  read the checked-out branch (`0728-3`) while CI runs on `0728-2`, so it found
  zero runs and exited 0 looking healthy.

- The link checker now scans six markdown trees instead of one (781 -> 921 files,
  30 -> 74 URLs actually checked). The extractor was fixed **first**, so widening
  imported real signal rather than 18 extraction artifacts: placeholder patterns
  now run against the raw capture, because cleaning turns
  `https://media.rediacc.com/...` into a plausible URL that 404s. Its allowlist
  also gained a liveness audit that fails offline on a dead entry and only warns
  on a network probe, since 9 of 14 entries answer 2xx from a residential IP.

- New `Review Complete` check, landed **non-required** and with no ruleset change.
  `Review Gate` could never see the review it names (Claude Review starts only
  after CI is green) and is SHA-agnostic, so a reply to an older round satisfies
  it for an arbitrary head. The new check is event-driven, posts against the
  PR's current head, and is acyclic by construction - a guard fails the build if
  any other workflow references the context, which I re-planted myself to
  confirm.

- The widened link checker found a defect in **itself** on its first CI run: a
  HEAD that hangs threw past the existing GET fallback, so a host that simply
  does not serve HEAD (`azure.microsoft.com/pricing/calculator/`, measured HEAD
  404 x3 / GET 200) was reported as a dead link. A URL is now only broken once
  GET has failed too, controlled so the rescue cannot whitewash an unresolvable
  host.

- `azure.microsoft.com` allowlisted with the measurement written down: the page
  does not implement HEAD (404 x3 / GET 200 from a residential IP, which the new
  GET fallback fixes generally) **and** the host is unreachable from runners
  (root 200 in 6.3s here, TIMEOUT there). Only the first was fixable in code.

- **Ceph provisioning now fails fast with the real cause, instead of timing out
  generically ten minutes later.** `ops up` logged a failed Ceph provision as
  non-fatal and returned success, so the caller waited for a cluster that was
  never coming. renet (981a4a963, PR #95) records the failure and short-circuits
  `WaitForHealth` with a new exit 11 carrying the original error verbatim, e.g.
  `failed to install prerequisites on node 22: ssh command failed: signal:
  killed` (an OOM kill during apt). `ops up` still SUCCEEDS, so nothing that
  relies on it changes behaviour.

- **And the console half, without which the renet half achieves nothing.**
  `waitForCephHealth` only threw at its own 1200s budget, so an exit 11 fell
  through to the retry and a three-second diagnosis still cost twenty minutes.
  Exit 11 now breaks out immediately with renet's stderr; every other non-zero
  code still retries, because "not up yet" and "never built" are different
  states and only the second is hopeless.

- **The same call was killing renet one fifth of the way through its own wait.**
  renet's `CephHealthTimeout` is 600s and nothing in this repo sets the
  `CEPH_HEALTH_TIMEOUT` override, but the caller passed a hard 120000.
  `OpsCommandRunner`'s timeout path returns `code: -1`, which is not 0, so the
  loop read our own SIGTERM as "not healthy yet" and retried: the 1200s budget
  went on nine truncated attempts and the error finally raised was our own
  "Timeout exceeded" marker. It now gets the remaining budget. Swept the class:
  of the eight per-call ops timeouts, `ops_rustfs.go` has no internal waits at
  all and ceph provision already passed 600000, so this was the only one.

- **One racy E2E assertion, three red signals.** `E2E K8s Multinode` test 5b
  read the post-eviction pod state once, and run 30446348669 caught it as
  `web-... 0/1 ContainerCreating 3m49s ... rediacc11`: on the right node, still
  mounting its RBD volume. It now polls, matching the file's other
  `podRunningOn` call. Worth naming because the file runs serially, so that one
  line also skipped tests 6 and 7 and tripped the zero-skip gate, and because
  E2E is on the watchdog retry allowlist so a green second attempt would have
  hidden it.

- **The provision-record wrappers stopped saying the same thing twice.** The
  local battery (`npm run ci` -> `check:ci-renet` -> `i18n.sh`) flagged five new
  hardcoded strings, all from the ceph commit above. They are not translated:
  both helpers are unexported and every caller already supplies the human
  sentence, so the inner prose produced `could not clear the Ceph provision
  record: failed to remove /x: permission denied` -- the same thing twice. The
  wrappers now carry only the subject; the caller keeps the verb. Proven with a
  control: planting one prose string back makes the gate exit 1 and name it.

- **`Migration Test` moved from the deterministic set to the flaky set, in the
  watchdog's own test.** Round 21 went red on one assertion, `a migration failure
  must fail fast`, which is the retry-allowlist change colliding with a test
  written before it. Resolved in the test's terms: Stage Artifacts and the
  Docker/CLI builds are deterministic, so a blind retry buys nothing; Migration
  Test drives six live D1 clones against Cloudflare's API and fails for reasons
  the code cannot control. A Cloudflare 7500 on the sixth clone took 23 green
  jobs down with it. Moved rather than deleted, so the retry side now asserts it
  DOES retry and the anti-vacuity check covers the pattern list.

- **Stop-gate v5, and the autopilot's no-bypass invariant.** The gate watched
  the wrong queue: zero open worklist items all night while the harness Task
  list carried six pending, so every idle stop was legal. It now reads the task
  list, requires a `## Remaining` section in the last message, requires a
  `DEFAULT:` on every deferral, requires a session brief (six sessions share
  this worktree), and asks haiku on the quiet stops. `MAX_BLOCKS` is gone: a
  failed judge blocks. Measured: `claude -p` fires the project's Stop hook, and
  `--settings '{"hooks":{}}'` does not suppress it, so an env guard is the only
  thing standing between this design and infinite recursion.

- **A test nothing invoked, caught by the dead-bash gate.** Round 23 went red on
  `Quality / Code` and `Quality / Security`, both from one finding: the new stop-gate
  suite was never referenced by any tracked file. Fixed by running it from
  `.claude/hooks/test-hooks.sh` rather than allowlisting it, so its 21 cases now
  execute in CI (82/82 through the runner). The gate was right and the fix makes the
  tests real rather than silencing the complaint.

- **Loop, cron and compact continuity are now gated.** The loop schedule is declared
  and an overdue declaration blocks, which is how a dead loop gets noticed; only one
  cron is allowed; the `## Remaining` section must name every open task by id so it
  matches the operator's own view; and a handover document is required, handed back
  by a new `PostCompact` hook as `additionalContext`. Measured first: cron state
  exists nowhere on disk, so it has to be declared rather than discovered.

- **The Stop gate now reads the harness event instead of reconstructing it, and
  the design docs are enforced.** `SessionStart` hands `docs/ci-overhaul/` to a new
  session as context; the Stop hook blocks when the code has moved more than ten
  commits past those docs (it was 44, now reconciled in `06-progress.md`); the loop
  is derived from `session_crons` rather than declared; and a publish-divergence
  check catches both a remote that moved without us and a stale local branch
  sharing the publish name (this repo had one, 51 commits behind with zero unique
  commits). 33 cases, 94/94 through `.claude/hooks/test-hooks.sh`.

- **The review budget is sized to the diff, and the reviewer no longer spawns
  agents nothing waits for.** 3 passes up to 10k changed lines, 5 up to 50k, 7
  above, with `X/Y` carrying the size that produced it. The tier table lives in
  `.ci/scripts/lib/common.sh` because `review-status.sh` used to sed-parse the cap
  out of the gate script, and its own header warns that the two disagreeing
  resurrects the deadlock it exists to prevent. Separately, a review run had
  spawned three background agents, run out of turns, and posted "waiting on the
  three background review agents" as its report: full cost, zero output. A review
  has a fixed turn budget and no join point, so `--disallowed-tools Task,Agent`.

- **The Stop hook now catches a stale PR body before CI spends a round on it.**
  This PR lost a round to `Quality / Static`'s freshness gate because a push
  followed the last body edit. The hook compares the published ref's tip against
  the PR's `lastEditedAt` and blocks locally. It found its own bug on first run:
  `lastEditedAt` is not a valid field on `gh pr list` or `gh pr view`, and because
  an unreadable lookup reports "THIS IS A HOOK BUG" rather than passing, that
  surfaced immediately instead of becoming a permanent no-op. GraphQL now.

- **The remaining list must now say what is moving, and agree with the task list.**
  Every item declares ongoing/pending/blocked, and claiming "ongoing" for a task the
  harness still shows as pending blocks with the mismatch named. A "nothing is
  in_progress" nag was built and then deliberately dropped: a queue that is honestly
  parked is a legitimate state, and the agreement check already catches the failure
  that matters.

- **CLAUDE.md's account-key instruction pointed at a shape of solution that cannot
  exist.** It told readers to `curl` a 404 endpoint for the production ed25519 key.
  CI never needed it: `ACCOUNT_ED25519_PUBLIC_KEY` is already an organisation
  secret. Locally the value must be pasted, because GitHub secrets are write-only,
  so swapping one URL for another would have preserved the mistake. The `-fsS`
  warning is kept and generalised, since piping an unchecked HTTP response into
  that variable bakes a 404's HTML into `keys.ProductionPublicKey`.

## For the reviewer

- The submodule pointers here are **branch** commits, not main commits. renet
  and elite must land first and these pointers be re-pointed at the squash
  commits before this merges.
- `backup_push`/`backup_pull` are registered but not licence-gated. Proven
  live. The operator chose to correct the marketing copy rather than gate them,
  and that copy fix is in this branch.
- 154 functions carry `pending: true` tiers. That is deliberate and visible,
  not a silent default. A draft proposal exists for the decisions.

## Known red: `check:ci-tutorial-caption-sync`

This branch lands with that gate red on 24 tutorial/language combos, deliberately, and it is
not this branch's defect. The gate fetches **published** `words.json` from media.rediacc.com,
not from this tree.

Root cause is `packages/www/scripts/lib/vtt-emit.ts:264-277`: per cue, when aligned words
cover less than `MIN_WORD_TIMING_COVERAGE` (0.5) of that cue's text, the real timings are
discarded and estimated. The aligner maps words by character overlap, so Arabic, Russian and
CJK cross that threshold far more often, which matches the distribution exactly (ar 9, ru 8,
zh 4, then pt/ja/fr one each).

Both obvious remedies were measured and rejected: re-aligning all 24 pairs succeeded 24/24 and
they still fail, and the published bytes are identical to the local ones so republishing
changes nothing. The CDN purge has already been run.

This is a pre-existing caption-quality limitation that the TTS migration **exposed** rather
than caused; the gate had been passing on pre-migration data served from a stale cache. Full
analysis and the three options in `docs/caption-sync-flat-cues.md`.

**Last pushed:** `89eb77dc8`

- `89eb77dc8` docs(claude): teach the worklist verbs, not the markdown file
- `269f7b0d4` feat(hooks): the stop report is derived from the store, not from memory
- `18b4b844d` feat(hooks): worklist v10, event-sourced store with verified worker liveness
- `75fdd5f73` docs(ci-overhaul): second full green, and the reduced-run demonstration
- `0b04a6f38` feat(hooks): hand over CI facts per job, and fix a control that proved nothing

### CODE REVIEW

**Comment by mfbayraktar** [comment]
Re: review report 5102557647 -- per-finding response.

**The report contains no findings to address, because the review did not finish.** Its own progress list shows six of nine deep-reads still open (`scope-engine/scope-map/skip-plan-reconcile`, the gate scripts, `watchdog-monitor`, the workflow YAML, the remaining gates) and the final item, "Synthesize findings and post summary comment", unchecked. It posted zero inline comments and opened zero review threads, which is consistent with never reaching synthesis rather than with a clean bill of health.

So there is nothing here I can accept, defer or disagree with on the merits. Recording that plainly rather than replying "no findings, all good", which would misrepresent an unfinished review as a passing one.

**Two defects the report surfaced anyway, both now fixed:**

1. **The cost line named the wrong model.** It read `(claude-haiku-4-5-20251001)` while the action is invoked with `--model claude-sonnet-5`, which reads as "the flag is ignored". It is not: the line used `modelUsage | keys | first`, reporting one model by arbitrary object-key order, and the action bills a small model for its own bookkeeping. The run log for 30348055944 shows `claude-sonnet-5` five times and no haiku at all. Fixed in `f95533298`: every model is now listed by output-token share, so the comment can answer the question it appears to answer. That is issue #539's cosmetic half.

2. **An incomplete review still wrote its `claude-reviewed:` marker**, so the dedup machinery now treats that commit as reviewed. The marking guard checks that output posted, not that the review reached a verdict, so a run that stops early is indistinguishable from one that found nothing. Not fixed here: it is a review-pipeline design change, logged for the autopilot work rather than smuggled into this PR.

**What changed since the reviewed SHA** (`a196417`), all with the battery green and each control-proven:

- `2cf624370` -- #533 answered by deletion rather than merging: stack 1's `web:` service was never started and named an image CI does not publish, so it was unrunnable rather than merely unused. Removing it dissolves both the `:80` collision and an undocumented network-name collision with `private/elite`. Verified by planting a bogus var and confirming `check-compose-env` still fires.
- `f95533298` -- the model attribution above.

A re-review of the current head is welcome and would be the first one to actually read this diff.

File: .ci/scripts/private/license-e2e.sh [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+
+# install_license <repo> — moves the staged blob into the real license store.
+install_license() {
+    local repo="$1"
+    $SUDO rm -rf "${REPO_LICENSE_ROOT:?}/$repo"
+    $SUDO mkdir -p "$REPO_LICENSE_ROOT/$repo"
+    $SUDO cp "$WORK"/stage/*.json "$REPO_LICENSE_ROOT/$repo/"
+    # World-readable so the unprivileged renet invocations can read them, which
+    # is how a real install looks.
+    $SUDO chmod -R a+rX "$LICENSE_ROOT"
```
[the remark this answers is not in the document]
Comment (mfbayraktar) [reply .ci/scripts/private/license-e2e.sh]: Confirmed against the artifact before accepting it, and fixed in af53749c2.

Verified: `LICENSE_ROOT=/var/lib/rediacc/license` and `REPO_LICENSE_ROOT=$LICENSE_ROOT/repos` (lines 73-74), 11 `install_license` call sites, and the "does not disturb them" guarantee at line 30. All three of your claims hold.

Fix is narrower than a straight swap, because a bare `chmod -R a+rX "$REPO_LICENSE_ROOT/$repo"` would have broken traversal: unprivileged renet has to walk both parents to reach that directory. So the two parents get `chmod a+rX` **non-recursively** (directory bits only, cannot touch any other repo's contents) and the repo directory itself keeps the recursive chmod.

File: .ci/scripts/ci/skip-plan-reconcile.cjs [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+        // reusable workflow is skipped, GitHub materialises only the caller
+        // ("OPS Tests") and never its inner jobs ("OPS Tests / OPS Provision
+        // (linux-amd64)"), so a table keyed on inner names matches nothing and
+        // this would report planned-job-missing for what is really a skip.
+        //
+        // Found by the shadow reconciler on run 30388401305, which is exactly
+        // the false-fire it was built to surface before the gate went live.
+        // Reported as planned-run-but-skipped, the same finding a directly
+        // skipped leaf produces, because that is what actually happened.
+        const callers = [...new Set(expected.map((e) => e.split(' / ')[0]))];
```
[the remark this answers is not in the document]
Comment (mfbayraktar) [reply .ci/scripts/ci/skip-plan-reconcile.cjs]: Confirmed against the artifact before accepting it, and fixed in af53749c2.

Verified: `package-tests` is a flat top-level job at `ci.yml:572-573` with `needs: [initialize, quality]`, and the `tests` caller's display name is exactly `Tests + Infra` at `ci.yml:673-674`. The collision is real.

Rather than hand-list the exemption, I derived it: parsing `ci.yml`'s top-level jobs for `uses: ./.github/workflows/` classifies all eleven jobs whose display name carries a ` / `. Ten are genuine reusable callers; exactly one, `package-tests`, is flat. `FLAT_JOB_KEYS` holds that one key, and a load-time check rejects an entry that stops naming a real plan key so it cannot rot into a dead suppression.

Your second point was the more useful one: the branch was untested, and every existing case only mutated a conclusion rather than removing a job entry. Case 33 now removes the job with a skipped lookalike present, and carries a control that a genuine reusable caller still produces the caller-skip diagnosis, so the test cannot be satisfied by deleting caller derivation outright. Proven by planting: reverting the fix in place makes case 33 fail with exactly the misattribution you described, and the file was restored and checked byte-identical by sha256 before commit.
